src/amo/utils.js

```diff
@@ -0,0 +1,32 @@
+import { LANDING_PAGE_ADDON_COUNT } from 'amo/constants';
+import { getLanding, loadLanding, failLanding } from 'amo/actions/landing';
+import { featured as featuredAPI, search } from 'core/api';
+
+
+export function fetchLandingAddons({ addonType, api, dispatch }) {
+  dispatch(getLanding({ addonType }));
+
+  const filters = { addonType, page_size: LANDING_PAGE_ADDON_COUNT };
+  const landingRequests = [
+    featuredAPI({ api, filters }),
+    search({ api, filters: { ...filters, sort: 'rating' }, page: 1 }),
+    search({ api, filters: { ...filters, sort: 'hotness' }, page: 1 }),
+  ];
+
+  return Promise.all(landingRequests)
+    .then(([featured, highlyRated, popular]) => dispatch(
+      loadLanding({ addonType, featured, highlyRated, popular })
+    ))
+    .catch(() => dispatch(failLanding({ addonType })));
+}
+
+export function loadLandingAddons({ store: { dispatch, getState }, params }) {
+  const state = getState();
+  const addonType = params.pluralAddonType.replace(/s$/, '');
+
+  if (!(addonType === state.landing.addonType && !state.landing.loading)) {
```

Comment:
Yeah this is actually too much, modelled in my head off the more complex checks for search.

In this case `addonType !== state.landing.addonType` is fine though that would make the page eventually stale... though this data isn't nearly as changed as even search results.

Basically this shouldn't be here I think... I'll remove it.